src/ResourceManager/Dns/Commands.Dns/Zones/NewAzureDnsZone.cs

```diff
@@ -35,10 +43,30 @@ public class NewAzureDnsZone : DnsBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string ResourceGroupName { get; set; }
 
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The type of the zone, Public or Private. This property cannot be changed for a zone.")]
+        [ValidateNotNullOrEmpty]
```

Comment:
Please add a validateset here if the only optional allow are public and private.